File: r/word-count/word-count.R
library(magrittr)

word_count <- function(phrase) {
  # m <- gregexpr("[[:alnum:]]+", phrase)
  # as.list(table(regmatches(tolower(phrase), m)))

  return(phrase
     %>% gregexpr("[[:alnum:]]+", .)     # find the words
     %>% regmatches(tolower(phrase), .)  # extract them
     %>% table                           # count them
     %>% as.list)
}

File: julia/word-count/word-count.jl
function wordcount(sentence::AbstractString)::Dict{AbstractString, Integer}
    re = r"\b\w(?:[[:alnum:]']*\w)?\b"

    ## first take
    ## # extract the words from the sentence
    ## words = [ m.match for m in eachmatch(re, lowercase(sentence)) ]
    ## # create the histogram: this is not very efficient
    ## Dict([ (w, count(ww -> w == ww, words)) for w in words ]) 

    histogram = Dict{AbstractString, Integer}()
    for m in eachmatch(re, lowercase(sentence))
        word = m.match
        histogram[word] = get(histogram, word, 0) + 1
    end
    histogram

end
